launcher/ui/MainWindow.cpp

```diff
@@ -335,6 +338,14 @@ class MainWindow::Ui
         actionSettings->setShortcut(QKeySequence::Preferences);
         all_actions.append(&actionSettings);
 
+        actionUndoTrashInstance = TranslatedAction(MainWindow);
+        connect(actionUndoTrashInstance, SIGNAL(triggered(bool)), MainWindow, SLOT(undoTrashInstance()));
+        actionUndoTrashInstance->setObjectName(QStringLiteral("actionUndoTrashInstance"));
+        actionUndoTrashInstance.setTextId(QT_TRANSLATE_NOOP("MainWindow", "&Undo Last Instance Deletion"));
+        actionUndoTrashInstance->setEnabled(true);
```

Comment:
```suggestion
        actionUndoTrashInstance->setEnabled(APPLICATION->instances()->trashedSomething());
```

this seems to fix the issue